### PR TITLE
LOG-91 mapping of fields to columns; file with last sent

### PR DESCRIPTION
**Pull request by piro314**

### CODE REVIEW

File: src/main/java/com/logsentinel/agent/DatabaseAgent.java [hunk 1]
```diff
@@ -1,98 +1,232 @@
...
+            logger.error("Failed to create json payload", e);
+        }
+    }
+
+    private void prepareAndSend(List<Map<String, String>> rows) throws ApiException, JsonProcessingException {
+        List<BatchLogRequestEntry> batch = new LinkedList<>();
+        for (Map<String, String> row : rows) {
+            BatchLogRequestEntry entry = new BatchLogRequestEntry();
+
+            String actorDisplayName = row.remove(actorDisplayNameColumn);
```
[the remark this answers is not in the document]
Comment (piro314) [reply src/main/java/com/logsentinel/agent/DatabaseAgent.java]: not to duplicate them in the details payload

[the remark this answers is not in the document]
Comment (Glamdring) [reply src/main/java/com/logsentinel/agent/DatabaseAgent.java]: Add a comment explaining that :)

File: src/main/java/com/logsentinel/agent/DatabaseAgent.java [hunk 2]
```diff
@@ -1,98 +1,244 @@
...
+        } catch (IOException e) {
+            logger.error("Exception while trying to read from file. Can't guarantee consistency. Exiting", e);
+            System.exit(1);
+        }
+        return lines.size() > 0 ? lines.get(0) : "0";
+    }
+
+    private String prepareQuery(String query){
+        if(!query.contains("where")){
+            return query + " where "+queryCriteriaColumn + " > ? ";
```
Comment (Glamdring) [inline src/main/java/com/logsentinel/agent/DatabaseAgent.java]: whitespaces

File: src/main/java/com/logsentinel/agent/DatabaseAgent.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Glamdring) [reply src/main/java/com/logsentinel/agent/DatabaseAgent.java]: These fields are sent separately (as path variables), so we want to remove them from the "row", which is then serialized and sent as details (body)

**Pull request merged**